src/condor/htcondor_helper.py

```diff
@@ -2,174 +2,332 @@
+            # Force token authentication 
             htcondor.param["SEC_TOKEN_AUTHENTICATION"] = "REQUIRED"
             
-            self.collector = htcondor.Collector(self.config["collector_host"])
-            submit_host_url = self.config["submit_host"]
-            submit_host = htcondor.classad.quote(submit_host_url)
+            self.logger.info("HTCondor security parameters configured")
+            
+            # Setup collector
+            self.collector = htcondor.Collector(self.config.collector_host)
+            submit_host = htcondor.classad.quote(self.config.submit_host)
             
-            # Setup security manager with token
+            # Query scheduler daemon
+            self.logger.debug(f"Querying for schedd on {self.config.submit_host}")
             schedd_ads = self.collector.query(
                 htcondor.AdTypes.Schedd,
                 constraint=f"Name=?={submit_host}",
                 projection=["Name", "MyAddress", "DaemonCoreDutyCycle", "CondorVersion"]
             )
 
             if not schedd_ads:
-                raise ValueError(f"No scheduler found for {submit_host_url}")
+                raise ValueError(f"No scheduler found for {self.config.submit_host}")
             
             schedd_ad = schedd_ads[0]
+            self.logger.debug(f"Found scheduler: {schedd_ad.get('Name', 'Unknown')}")
             self.schedd = htcondor.Schedd(schedd_ad)
 
+            # Test connection to schedd
+            self.logger.debug("Testing connection to schedd...")
+            try:
+                test_query = self.schedd.query(constraint="False", projection=["ClusterId"])
+                self.logger.debug(f"Successfully connected to schedd, got {len(test_query)} results")
+            except Exception as e:
+                self.logger.error(f"Failed to connect to schedd: {str(e)}")
+                raise
+            
+            # Query credential daemon
+            self.logger.debug(f"Querying for credential daemon on {self.config.submit_host}")
             cred_ads = self.collector.query(
                 htcondor.AdTypes.Credd,
-                constraint=f'Name == "{submit_host_url}"'
+                constraint=f'Name == "{self.config.submit_host}"'
             )
 
             if not cred_ads:
-                print(f"No credential daemon found for {submit_host_url}. Continuing without it.")
+                self.logger.warning(f"No credential daemon found for {self.config.submit_host}. Continuing without it.")
                 self.credd = None
             else:
                 cred_ad = cred_ads[0]
-                print(f"Found credential daemon: {cred_ad.get('Name', 'Unknown')}")
+                self.logger.debug(f"Found credential daemon: {cred_ad.get('Name', 'Unknown')}")
                 self.credd = htcondor.Credd(cred_ad)
                 
                 # Add credentials for required services
-                self._add_credentials()
+                if self.credd:
```

Comment:
I am pretty sure this section of code is duplicated several times throughout this file. it would be best to make it into a function that can be called. (please review the code that this PR removes)